runtime/tests/automl_runtime/forecast/utils_test.py

```diff
@@ -165,3 +167,65 @@ def test_generate_cutoffs_success_annualy(self):
         ).rename_axis("y").reset_index()
         cutoffs = generate_cutoffs(df, horizon=1, unit="YS", num_folds=3, seasonal_period=1)
         self.assertEqual([pd.Timestamp('2018-07-14 00:00:00'), pd.Timestamp('2019-07-14 00:00:00'), pd.Timestamp('2020-07-14 00:00:00')], cutoffs)
+
+class TestCalculatePeriodsAndFrequency(unittest.TestCase):
+    def setUp(self) -> None:
+        return super().setUp()
+    
+    def test_calculate_periods_evenly(self):
+        start_time = pd.Series(
+            ['2021-01-14', '2021-02-14', '2021-03-14']
+        )
+        end_time = pd.Series(
+            ['2021-05-14', '2021-07-14', '2022-03-14']
+        )
+        periods = calculate_periods(
+            start_time, end_time, 'month'
+        )
+        self.assertTrue((periods == pd.Series([4, 5, 12])).all())
+        # self.assertTrue(consistency)
+    
+    def test_calculate_periods_unevenly(self):
+        start_time = pd.Series(
+            ['2021-01-14', '2021-02-14', '2021-03-14']
+        )
+        end_time = pd.Series(
+            ['2021-05-12', '2021-07-15', '2021-03-14']
+        )
+        periods = calculate_periods(
+            start_time, end_time, 'month'
+        )
+        self.assertTrue((periods == pd.Series([4, 5, 0])).all())
+        # self.assertFalse(consistency)
+    
+    def test_scalar(self):
+        start_time = pd.Series(
+            ['2021-01-14', '2021-02-14', '2021-03-14']
+        )
+        end_time = pd.Series(
+            ['2021-05-14', '2021-07-14', '2022-03-14']
+        )
+        start_scalar = pd.to_datetime('2021-01-14')
+        end_scalar = pd.to_datetime('2021-05-14')
+        self.assertTrue(
+            (calculate_periods(start_scalar, end_scalar, 'month') == \
+                pd.Series([4])).all())
+        self.assertTrue(
+            (calculate_periods(start_scalar, end_time, 'month') == \
+                pd.Series([4, 6, 14])).all())
+        self.assertTrue(
+            (calculate_periods(start_time, end_scalar, 'month') == \
+                pd.Series([4, 3, 2])) .all())
+
+    def test_frequency_consistency(self):
```

Comment:
Could also we add a/some test case(s) for non-monthly freq for `is_frequency_consistency`?